app/controllers/organizations/pets_controller.rb

```diff
@@ -29,11 +29,21 @@ def show
   end
 
   def create
+    transaction_success = false
     @pet = Pet.new(pet_params)
 
-    if @pet.save
+    ActiveRecord::Base.transaction do
+      if @pet.save && Organizations::DefaultPetTaskService.new(@pet).create_tasks
```

Comment:
Instead of doing this `transaction_success` management you can do raise within the block and rescue that then call `ActiveRecord::Rollback`

```
ActiveRecord::Base.transaction do
   begin
      @pet.save! 
      Organizations::DefaultPetTaskService.new(@pet).create_tasks!
   rescue
      raise ActiveRecord::Rollback
   end
end
```
Any raised error would trigger the rollback.